packages/tools/fluid-runner/src/logger/csvFileLogger.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as fs from "fs";
+import path from "path";
+import readline from "readline";
+import { v4 as uuidv4 } from "uuid";
+import { AsyncParser } from "json2csv";
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { BaseFileLogger } from "./baseFileLogger";
+
+/**
+ * FileLogger that writes events into a defined CSV file
+ * @internal
+ */
+export class CSVFileLogger extends BaseFileLogger {
+    /** Store the column names to write as the CSV header */
+    private readonly columns = new Set();
+
+    /** Store the telemetry in a temporary file to convert to CSV later */
+    private readonly tempFile = path.join(__dirname, uuidv4());
+
+    public async flush(): Promise<void> {
+        return super.flushCore(this.tempFile, "\n");
```

Comment:
It feels like "flush" does not make any sense for this logger. I.e., flush() is mostly useful to ensure that not all data is lost in case of application crash. Memory usage is secondary here.
With CSV flow we can flush temporary state, but it's not very useful as nobody can use it. If if we are fine with it, it's not clear if writing temp state to disk makes sense - I'd think amount of data we operate here is low and it makes much more sense just to keep it in memory.